lib/ssz_ex.ex

```diff
@@ -64,6 +77,95 @@ defmodule LambdaEthereumConsensus.SszEx do
     |> flatten_results_by(&Enum.join/1)
   end
 
+  defp decode_bitlist(bit_list, max_size) do
+    num_bytes = byte_size(bit_list)
+
+    with {:ok, len} <- get_bitlist_len(bit_list) do
+      cond do
+        div(len, 8) + 1 != num_bytes ->
+          {:error, "InvalidByteCount"}
+
+        len > max_size ->
+          {:error, "OutOfBounds"}
+
+        true ->
+          bytes_for_bit_len = max(1, div(len + 7, 8))
+          bit_size = bit_size(bit_list)
+          index = rem(len, 8)
+          skip = bit_size - index - 1
+          <<pre::bitstring-size(skip), _::size(1), rest::bitstring>> = bit_list
+
+          <<pre::binary-size(bytes_for_bit_len - 1), rest::bitstring>> =
+            <<pre::bitstring, 0::1, rest::bitstring>>
+
+          remain_bits_size = len - bit_size(pre)
+          <<_::size(8 - remain_bits_size), rest::bitstring>> = rest
+          formatted = <<pre::bitstring, rest::bitstring>>
+
+          if byte_size(formatted) > bytes_for_bit_len do
+            <<truncated::binary-size(bytes_for_bit_len), _rest::bitstring>> = formatted
+            {:ok, truncated}
+          else
+            {:ok, formatted}
+          end
+      end
+    end
+  end
+
+  defp get_bitlist_len(bit_list) do
+    :binary.bin_to_list(bit_list)
+    |> Enum.with_index()
+    |> Enum.reverse()
+    |> Enum.find_value({:error, "missing_length_information"}, fn {byte, byte_index} ->
+      if for(<<(bit::1 <- <<byte>>)>>, do: bit) |> Enum.any?(&(&1 == 1)) do
+        {:ok, {byte, byte_index}}
+      else
+        {:error, "missing_length_information"}
+      end
+    end)
+    |> case do
+      {:ok, {byte, byte_index}} ->
+        leading_zeros =
+          for(<<(bit::1 <- <<byte>>)>>, do: bit)
+          |> Enum.take_while(&(&1 == 0))
+          |> Enum.count()
+
+        {:ok, byte_index * 8 + 7 - leading_zeros}
+
+      error ->
+        error
+    end
+  end
+
+  defp decode_bitvector(bit_vector, size) when bit_size(bit_vector) == size,
+    do: {:ok, BitVector.new(bit_vector, size)}
+
+  defp decode_bitvector(_bit_vector, _size), do: {:error, "invalid bit_vector length"}
+
+  defp encode_bitlist(bit_list, max_size) do
+    len = bit_size(bit_list)
+    bytes_len = byte_size(bit_list)
+    bytes_for_bit_len = max(1, div(len + 1 + 7, 8))
+
+    if len > max_size do
```

Comment:
Instead of nested `ifs`, we can use `cond`:

```elixir
cond do
  len > max_size -> {:error, "Bitlist of size #{len}, but has a max of #{max_size}"}
  bytes_for_bit_len > bytes_len -> ...
  true -> ...
end
```